(#232,#235,#236,#240) .NET SDK v3.0.2 items

adrianhall opened the pull request.

azurecla commented:
Hi __@adrianhall__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (adrianha). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

adrianhall commented on README.md:
```diff
@@ -13,7 +13,18 @@ To get the source code of our SDKs and samples via **git** just type:
-## Reference Documentation
+## Supported platforms
+
+* Xamarin Android releases for API 19 through 24 (KitKat through Nougat)
```
I don't think we can say API 19+, as API 24 proved.  I'd rather do the work on each new version and have to do a quick README update.

adrianhall commented on README.md:
```diff
@@ -13,7 +13,18 @@ To get the source code of our SDKs and samples via **git** just type:
+* Xamarin iOS releases for iOS versions 8.0 through 10.0
+* Universal Windows Platform
+* Windows Phone 8.1
+* Windows Phone 8.0 except for Silverlight applications.
```
Duh - yeah - will correct.  Thanks.

dhei replied:
yeah, it's Windows Phone Silverlight 8.1

adrianhall commented on README.md:
```diff
@@ -13,7 +13,18 @@ To get the source code of our SDKs and samples via **git** just type:
+## Supported platforms
+
+* Xamarin Android releases for API 19 through 24 (KitKat through Nougat)
+* Xamarin iOS releases for iOS versions 8.0 through 10.0
```
As with the Android version, unless Apple is committing to testing our library, we need to ensure compatibility before we support it.

adrianhall commented:
Re: the PCLStorage.  If I do the System.IO, I have to add a Platform specific IO package in.  I was going along the same lines.  However, SQLiteStore is a PCL, so it doesn't get the entire System.IO - most notably, File.* are not there, so I can't do File.Create or File.Exists.  

The way around this is to add another method (EnsureDatabaseExists(fullpathname)) to the IPlatform.cs and do it there.  That would work around the issue.  I'll discuss internally on Monday and see which direction we want to go.  (@dhei - if you have comment over the weekend, please add it)

jamesmontemagno commented:
I would most likely just want to validate support from @dplaisted on it. It think that going against System.IO and controlling it is worth it as taking a dependency on PLCStorage brings in even more DLLs, I think it would be better to bring in platform specific DLLs to access System.IO and actually for 4.0 they will go away with .NET Standard

dhei commented:
@adrianhall: I agree with @jamesmontemagno. I think it's easier to manage platform specific IO package than adding a new PCL package. More importantly, [PCLStorage](https://www.nuget.org/packages/pclstorage) takes dependency on Microsoft.Bcl >= 1.1.6 but we're planning to completely get rid of Microsoft.Bcl when we moves to .NET Standard 1.3.

adrianhall commented on src/Microsoft.WindowsAzure.MobileServices.SQLiteStore/MobileServiceSQLiteStore.cs:
```diff
@@ -44,8 +45,27 @@ public MobileServiceSQLiteStore(string fileName)
+                }
+                catch (FileNotFoundException)
+                {
+                    // Ignore this error - it means the file exists
```
Ok - irrelevant now since I've factored it out.

adrianhall commented:
Per the review comments, I've added a per-platform method called "EnsureFileExists(path)".  On Windows, it uses Windows.Storage (since System.IO.File does not exist - go figure), and on iOS/Android, it uses System.IO.File.  I've also removed the dependency on PCLStorage from the SQLiteStore. 

This does increase our public API footprint on the MobileServiceClient, but I think both pieces are likely to be useful for anyone implementing a local store provider, so it's a good addition.  

Comments appreciated.

dhei commented on Microsoft.WindowsAzure.Mobile.Managed_Full_IncludeXamarin.sln:
```diff
@@ -1,6 +1,6 @@
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio 2013
-VisualStudioVersion = 12.0.40629.0
```
Microsoft.WindowsAzure.Mobile.Managed_Full_IncludeXamarin.sln is supposed work with Visual Studio 2013 and Microsoft.WindowsAzure.Mobile.Managed_Full_IncludeXamarin_V15.sln is for Visual Studio 2015. So can you make this change in V15 sln file?

dhei replied:
The rest looks good to me.

dhei commented:
>     public class CurrentPlatform : IPlatform

Please add EnsureFileExists() to this class.

Error CS0535 'CurrentPlatform' does not implement interface member 'IPlatform.EnsureFileExists(string)' Microsoft.WindowsAzure.Mobile.Ext.Android 


---
Refers to: src/Microsoft.WindowsAzure.MobileServices.Android/Platform/CurrentPlatform.cs:12 in b90a6d0. [](commit_id = b90a6d07b3e63d78674b0ab2b0280f8a463c9b8d, deletion_comment = False)
